### ISSUE TITLE
Misleading error when incorrectly using crate as a module.

### ISSUE THREAD

**Issue by iago-lito** [rust-lang/rust#68821]
I had a source file used as a module in my binary crate. At some point, I have decided to turn this file into a second binary crate instead. But I forgot to remove the corresponding `mod` statement in the first crate. It took me a while to figure it out because the error message was very misleading.

Here is the situation boiled down:

`Cargo.toml`
```toml
[package]
name = "rust_test"
version = "0.1.0"
authors = ["Iago-lito"]
edition = "2018"
default-run = "first_bin"

[[bin]]
name = "first_bin"
path = "src/first_bin.rs"

[[bin]]
name = "second_bin"
path = "src/second_bin.rs"
```

`src/first_bin.rs`
```rust
mod common_module;
mod second_bin; // That is very wrong..

fn main() {
    common_module::common_function();
} 
```
`src/second_bin.rs`
```rust
mod common_module; // .. but the error blames this line.

fn main() {
    common_module::common_function();
} 
```

`src/common_module.rs`
```rust
pub fn common_function() {}
```

I'd expect getting something like:
```
error[EXXX]: binary crate root file cannot be used as a module.
--> src/first_bin:2:5
  |
2 | mod second_bin;
  |     ^^^^^^^^^^
  |
  = help: remove this statement or consider not declaring `src/second_bin.rs` as the root of a binary crate in `Cargo.toml`
```

But the actual error is:
```
error[E0583]: file not found for module `common_module`
 --> src/second_bin.rs:1:5
  |
1 | mod common_module;
  |     ^^^^^^^^^^^^^
  |
  = help: name the file either second_bin/common_module.rs or second_bin/common_module/mod.rs inside the directory "src"
```

Because of this, I had to dig the whole way through [Chapter 7 of the Book](https://doc.rust-lang.org/book/ch07-00-managing-growing-projects-with-packages-crates-and-modules.html) again before I figured out that there was nothing wrong with the location of `src/common_module.rs`. Then I had to scroll the other files line by line to eventually spot the mistake.

Would this be considered a bug? Is it the correct place to report it?
